### ISSUE TITLE
Unsupported linux distribution

### ISSUE THREAD

**Issue by trendoid** [Microsoft/sqlopsstudio#2390]
Issue Type: <b>Bug</b>

search for extension,
click link to download,
install vsiz via command palette,
reload,
error message pops up

SQL Operations Studio version: sqlops 0.31.4 (e026ab85a72c190cc4ccc834503aa9fe36dd6407, 2018-07-18T01:56:25.508Z)
OS version: Linux x64 4.15.0-33-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-5200U CPU @ 2.20GHz (4 x 2038)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>video_decode: unavailable_software<br>video_encode: enabled<br>vpx_decode: unavailable_software<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 1, 1|
|Memory (System)|7.70GB (0.31GB free)|
|Process Argv|/usr/share/sqlops/sqlops --unity-launch|
|Screen Reader|no|
|VM|0%|

</details>

**Comment by trendoid** [comment]
![screenshot from 2018-08-31 14-01-54](https://user-images.githubusercontent.com/7746539/44928949-45223680-ad27-11e8-93e2-cde4bcb498c7.png)

**Comment by trendoid** [comment]
[2018-08-31 14:10:35.403] [exthost1] [info] extension host started
[2018-08-31 14:10:35.473] [exthost1] [info] ExtensionService#_doActivateExtension Microsoft.account-provider-azure {"startup":true,"activationEvent":"*"}
[2018-08-31 14:10:35.474] [exthost1] [info] ExtensionService#loadCommonJSModule /usr/share/sqlops/resources/app/extensions/account-provider-azure/out/main
[2018-08-31 14:10:36.324] [exthost1] [info] ExtensionService#_doActivateExtension vscode.git {"startup":true,"activationEvent":"*"}
[2018-08-31 14:10:36.324] [exthost1] [info] ExtensionService#loadCommonJSModule /usr/share/sqlops/resources/app/extensions/git/out/main
[2018-08-31 14:10:36.660] [exthost1] [info] ExtensionService#_doActivateExtension vscode.merge-conflict {"startup":true,"activationEvent":"*"}
[2018-08-31 14:10:36.660] [exthost1] [info] ExtensionService#loadCommonJSModule /usr/share/sqlops/resources/app/extensions/merge-conflict/out/extension
[2018-08-31 14:10:36.677] [exthost1] [info] ExtensionService#_doActivateExtension Microsoft.server-report {"startup":true,"activationEvent":"*"}
[2018-08-31 14:10:36.677] [exthost1] [info] ExtensionService#loadCommonJSModule /home/trendoid/.sqlops/extensions/Microsoft.server-report-0.1.3/out/src/extension
[2018-08-31 14:10:36.684] [exthost1] [info] ExtensionService#_doActivateExtension Microsoft.whoisactive {"startup":true,"activationEvent":"*"}
[2018-08-31 14:10:36.684] [exthost1] [info] ExtensionService#loadCommonJSModule /home/trendoid/.sqlops/extensions/Microsoft.whoisactive-0.1.1/out/src/extension
[2018-08-31 14:10:36.787] [exthost1] [info] eager extensions activated
[2018-08-31 14:10:36.794] [exthost1] [info] ExtensionService#_doActivateExtension vscode.configuration-editing {"startup":false,"activationEvent":"onLanguage:jsonc"}
[2018-08-31 14:10:36.794] [exthost1] [info] ExtensionService#loadCommonJSModule /usr/share/sqlops/resources/app/extensions/configuration-editing/out/extension
[2018-08-31 14:10:36.806] [exthost1] [info] ExtensionService#_doActivateExtension vscode.json-language-features {"startup":false,"activationEvent":"onLanguage:jsonc"}
[2018-08-31 14:10:36.806] [exthost1] [info] ExtensionService#loadCommonJSModule /usr/share/sqlops/resources/app/extensions/json-language-features/client/out/jsonMain
[2018-08-31 14:10:36.961] [exthost1] [info] ExtensionService#_doActivateExtension Microsoft.mssql {"startup":true,"activationEvent":"*"}
[2018-08-31 14:10:36.961] [exthost1] [info] ExtensionService#loadCommonJSModule /usr/share/sqlops/resources/app/extensions/mssql/out/main
[2018-08-31 14:10:37.286] [exthost1] [info] ExtensionService#_doActivateExtension Microsoft.agent {"startup":true,"activationEvent":"*"}
[2018-08-31 14:10:37.286] [exthost1] [info] ExtensionService#loadCommonJSModule /home/trendoid/.sqlops/extensions/Microsoft.agent-0.32.7/out/main
[2018-08-31 14:10:37.313] [exthost1] [info] ExtensionService#_doActivateExtension Microsoft.import {"startup":true,"activationEvent":"*"}
[2018-08-31 14:10:37.313] [exthost1] [info] ExtensionService#loadCommonJSModule /home/trendoid/.sqlops/extensions/Microsoft.import-0.1.0/out/main
[2018-08-31 14:10:37.558] [exthost1] [info] ExtensionService#_doActivateExtension Microsoft.profiler {"startup":true,"activationEvent":"*"}
[2018-08-31 14:10:37.558] [exthost1] [info] ExtensionService#loadCommonJSModule /home/trendoid/.sqlops/extensions/Microsoft.profiler-0.1.5/client/out/main

**Comment by kburtram** [comment]
@trendoid which Linux distro are you using?  We have some limitations based on what .Net Core 2.1 supports (details available at https://docs.microsoft.com/en-us/dotnet/core/linux-prerequisites?tabs=netcore2x).

**Comment by trendoid** [comment]
Ubuntu 18.04

**Comment by trendoid** [comment]
Do you need more exact version information?  I'm running a Dell XPS 13 from 2015 with Broadcom on it, so the kernel is built from a repo that allows for that driver.  I didn't get deep in the code to see how you check for version. I apologize.

**Comment by trendoid** [comment]
Is this an issue?

in: /home/trendoid/.sqlops/extensions/Microsoft.import-0.1.0/node_modules/service-downloader/out/platform.js

There is:

 case 'ubuntu':
            if (distributionVersion.startsWith('14')) {
                // This also works for Linux Mint
                return Runtime.Ubuntu_14;
            }
            else if (distributionVersion.startsWith('16')) {
                return Runtime.Ubuntu_16;
            }
            break;

**Comment by MattIrv** [comment]
Thanks for tracking this down. `service-downloader` was updated recently to support Ubuntu 18 but maybe there's a bug.

@trendoid If you look down a bit further in that `platform.js` file (around line 60), does the end of that case statement look like this?

```
        default:
            // Default to Ubuntu_16 to try to support other Linux distributions
            return Runtime.Ubuntu_16;
    }
    return Runtime.Ubuntu_16;
```

or like this?

```
        default:
            return Runtime.Unknown;
    }
    return Runtime.Unknown;
```

**Comment by MattIrv** [comment]
Thanks for the quick response. That code is supposed to take care of things for Ubuntu 18 but apparently isn't. I'll take a look and see if I can find a workaround

**Comment by MattIrv** [comment]
@trendoid does the rest of SQL Ops Studio work correctly for you? Are you able to connect to servers and execute queries?

**Comment by trendoid** [comment]
Yep. Works.  Just this one new extension.  I saw the release notes and got all giddy, but no go for me :-(

**Comment by MattIrv** [comment]
I'll have a PR out shortly to fix this.

@trendoid if you need a workaround you should be able to edit a config file to get everything working. The file will be `~/.sqlops/extensions/Microsoft.import-0.1.0/out/services/config.json` and you need to change "Linux_64" to "Ubuntu_16" on line 9 and then restart SQL Ops Studio.

I've also attached a version of the `config.json` file that you can replace the old one with and it should work (make sure to rename the attachment from `config.json.txt` to `config.json`), even for anyone who runs into this error who isn't on Ubuntu.

Let me know if you have any trouble with it, and thanks again for all your help tracking this down.

[config.json.txt](https://github.com/Microsoft/sqlopsstudio/files/2341755/config.json.txt)

**Comment by trendoid** [comment]
Thanks.  I'll update and try out the extension as soon as I can.  You guys are awesome as usual.

**Comment by MattIrv** [comment]
The second error was because we were accidentally uploading the Mac binaries as the Linux version. I've fixed that bug now and uploaded a new version of `config.json` above. If you replace your `config.json` file as described above then it should work now

**Comment by trendoid** [comment]
Thanks for all your help.  I'm testing the fix and it's downloading the linux version.  All seems well.  Thanks again.

**Comment by MattIrv** [comment]
@trendoid the stack trace you put there is from issue #2282. To work around it you can open the Import extension by right clicking on a database in Object Explorer and choosing "Import wizard". If it still doesn't work when you do that let me know
